### PR TITLE
Fix docker builds

### PR DESCRIPTION
**Pull request by adam-stokes**
This updates to a recent stable container for building snaps via docker.

Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>

### CODE REVIEW

File: snap/build-scripts/build [hunk 1]
```diff
@@ -27,7 +27,7 @@ if [ -z "$KUBE_SNAP_BINS" ]; then
             curl -LO \
               https://dl.k8s.io/${KUBE_VERSION}/bin/linux/$arch/$app
             chmod +x $app
-            if ! file ${app} 2>&1 | grep -q 'executable'; then
+            if [[ ! -x $app ]]; then
```
Comment (Cynerva) [inline snap/build-scripts/build]: This case will never occur 'cause it's checking for the executable bit, which we just set with `chmod +x` on the previous line.

The `file | grep executable` check has some history to it. I don't have time to pull up the history right this sec but I'm pretty sure we accidentally pushed out a snap whose "executable binary" was an XML file. We should be careful not to lose this if it's still protecting us from something.

**Review by kwmonroe — APPROVED** [review]
One day in the not-too-distant-future, i'd like this to be able to build arm/ppc/s390 snaps too.  But since we default to amd today, it lgtm as-is.

File: snap/docker-build.sh [hunk 2]
```diff
@@ -1,4 +1,13 @@
 #!/bin/bash
 
-docker run --rm -v $PWD:/root/snap -w /root/snap -e SNAPCRAFT_SETUP_CORE=1\
-    snapcraft/xenial-amd64 make "$@"
+docker run --rm -v "$PWD":/root/snap -w /root/snap \
+       -e SNAPCRAFT_SETUP_CORE=1 \
+       -e KUBE_ARCH=amd64 \
+       -e KUBE_VERSION=$(curl -L https://dl.k8s.io/release/stable.txt) \
+       -v /usr/bin/curl:/usr/bin/curl \
```
Comment (kwmonroe) [inline snap/docker-build.sh:7]: I'm sure the build system has all this stuff already, but to make it portable, perhaps this script should do something like:
```bash
if [ ! -e /usr/bin/[curl|make|etc] ]; then
  apt install [curl|make|etc]
fi
```

Comment (adam-stokes) [reply snap/docker-build.sh:7]: I have a ansible playbook when auto creating jenkins slaves, ill add this there.

**Pull request merged**